From fc9dfba9a4ff92e18acf05ac2dad9ca57a2fed48 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 14:58:05 -0400
Subject: [infiniband] usnic: Add struct usnic_transport_spec

Message-id: <e73fcd56c699c316f9f1a1b02e5fe359543b6ebe.1411569661.git.dledford@redhat.com>
Patchwork-id: 96574
O-Subject: [Patch RHEL7 03/29] IB/usnic: Add struct usnic_transport_spec
Bugzilla: 916382
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Upinder Malhi <umalhi@cisco.com>

Add *struct usnic_transport_spec* for passing around transport
specifications.

Signed-off-by: Upinder Malhi <umalhi@cisco.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 301a0dd68e5ddd22d992a58f466b621987d9df3b)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/usnic/usnic_abi.h b/drivers/infiniband/hw/usnic/usnic_abi.h
index 510b7d7..dd3e4c6 100644
--- a/drivers/infiniband/hw/usnic/usnic_abi.h
+++ b/drivers/infiniband/hw/usnic/usnic_abi.h
@@ -33,6 +33,15 @@ enum usnic_transport_type {
  USNIC_TRANSPORT_MAX  = 2,
 };
 
+struct usnic_transport_spec {
+ enum usnic_transport_type trans_type;
+ union {
+  struct {
+   uint16_t port_num;
+  } usnic_roce;
+ };
+};
+
 /*TODO: Future - usnic_modify_qp needs to pass in generic filters */
 struct usnic_ib_create_qp_resp {
  u32    vfid;
-- 
1.7.1